### PR TITLE
Refs #26135 - Allow hammer to send off content-type

### PR DESCRIPTION
**Pull request by chris1984**
Tests are failing because the Katello PR is not in yet, once that is in I will come back to this. Adding in now for a review from @akofink and then once the Katello PR is in I will get the tests fixed up etc. Setting to  Waiting on Contrib.

### CODE REVIEW

File: lib/hammer_cli_katello/repository.rb [hunk 1]
```diff
@@ -436,12 +435,14 @@ def update_content_upload(upload_id, repo_id, file)
...
       def import_uploads(uploads, opts = {})
+        rpm_type = option_content_type
         publish_repository = opts.fetch(:last_file, false)
         sync_capsule = opts.fetch(:last_file, false)
         params = {:id => get_identifier,
                   :uploads => uploads,
                   publish_repository: publish_repository,
-                  sync_capsule: sync_capsule
+                  sync_capsule: sync_capsule,
+                  content_type: rpm_type,
```
Comment (akofink) [inline lib/hammer_cli_katello/repository.rb]: Not sure why we need the `rpm_type` variable instead of just passing in `option_content_type`.

**Comment by chris1984** [comment]
@akofink  So looking at the test failures would you be opposed to an if option_content_type? here for this and else just pass in the params as normal?

https://github.com/Katello/hammer-cli-katello/blob/27f157975646be274fe83d68242e42d9df344d05/lib/hammer_cli_katello/repository.rb#L440

Tests are failing it looks like because we are expecting a param to be passed and since it is optional i think that is the best route but wanted to ask you first

**Comment by akofink** [comment]
I don't think that will fix your tests. You need to include the updated apipie cache.

**Comment by akofink** [comment]
I think this needs a rebase

**Comment by vijay8451** [comment]
@chris1984 I tried to fetch this one but not working due to conflict .

**Comment by chris1984** [comment]
> @chris1984 I tried to fetch this one but not working due to conflict .

Updated

**Comment by akofink** [comment]
You should probably only update the apipie cache for Katello 3.13. These tests aren't run against 3.12. https://github.com/Katello/hammer-cli-katello/blob/master/test/test_helper.rb#L20

**Comment by chris1984** [comment]
@akofink yep I am going to, also working on adding a test for the content type too,

**Pull request closed without merging**